addons/l10n_ro_efactura/controllers/main.py

```diff
@@ -26,31 +33,58 @@ def callback(self, company_id, **kw):
         """ Use the acquired access_key to request access & refresh token from ANAF """
         company = http.request.env['res.company'].browse(company_id)
         access_key = kw.get('code')
+
+        def log_and_raise_error(message: str):
+            message += '\n' + _("Received access key: %s", access_key)
+            company._l10n_ro_edi_log_message(message, 'callback')
+            raise UserError(message)
+
         # Without certificate, ANAF won't give any access key in the callback URL's "code" parameter
         if not access_key:
-            error_message = _("Access key not found. Please try again.\nResponse: %s", kw)
-            company.l10n_ro_edi_oauth_error = error_message
-            company.env.cr.commit()
-            raise UserError(error_message)
-
-        response = requests.post(
-            url='https://logincert.anaf.ro/anaf-oauth2/v1/token',
-            data={
-                'grant_type': 'authorization_code',
-                'client_id': company.l10n_ro_edi_client_id,
-                'client_secret': company.l10n_ro_edi_client_secret,
-                'code': access_key,
-                'access_key': access_key,
-                'redirect_uri': company.l10n_ro_edi_callback_url,
-                'token_content_type': 'jwt',
-            },
-            headers={
-                'content-type': 'application/x-www-form-urlencoded',
-                'accept': 'application/json',
-                'user-agent': 'PostmanRuntime/7.29.2',
-            },
-            timeout=10,
-        )
-        response_json = response.json()
-        company._l10n_ro_edi_process_token_response(response_json)
+            log_and_raise_error(_("Access key not found. Please try again.\nResponse: %s", kw))
+
+        try:
+            response = requests.post(
+                url=URL_ANAF_TOKEN,
+                data={
+                    'grant_type': 'authorization_code',
+                    'client_id': company.l10n_ro_edi_client_id,
+                    'client_secret': company.l10n_ro_edi_client_secret,
+                    'code': access_key,
+                    'access_key': access_key,
+                    'redirect_uri': company.l10n_ro_edi_callback_url,
+                    'token_content_type': 'jwt',
+                },
+                headers={
+                    'content-type': 'application/x-www-form-urlencoded',
+                    'accept': 'application/json',
+                    'user-agent': 'PostmanRuntime/7.29.2',
```

Comment:
tbh I'm not sure myself. This part was copied from Romanian OCA's code for making the requests, and it works. It might be because the API is strict about it because I see postman mentioned a lot in the API documentation

Reply:
Wait what? Copied from where?

Reply:
https://github.com/OCA/l10n-romania/blob/16.0/l10n_ro_account_anaf_sync/controllers/anaf_oauth.py#L122

Reply:
@yosuanicolaus that wasn’t really my point. You can’t copy people’s code without respecting the authorship and license (which is not compatible with our), saying “I copied code from OCA” is not ok. Did you only used that line or was there other code that was copied? If it’s the second case, you will need to rewrite it.

Reply:
@mart-e I get your point. I am respecting the authorship, the author of the romanian OCA were in contact with us, helped us through our development process, and specifically pointed to these part of the code, showed us their code for us to "copy", when we were having troubles trying to get the authentication request to be accepted by the API. The "copied" codes is just the request method data & headers as the official documentation was written very poorly on that part.

Is that not fine for you? Is it technically illegal and I should've lied about it? Or if it's fine, then should I've said something else? That this part of the code was "inspired" by the OCA? (sorry, I'm not too knowledgeable about legal stuffs)

Reply:
@yosuanicolaus no, if you discussed with the authors and they are helping, this is fine. It was not clear if by “copy” you went this specific line or was it the whole code bloc (which would be subject the OCA’s code licence and be problematic to include in our codebase).

> Is it technically illegal and I should've lied about it?

If, in the future, you want to do something illegal, I don’t recommend asking advices on a public website :stuck_out_tongue: 

Now to go back to xmo’s question, can you check if you need the agent to be exactly `PostmanRuntime/7.29.2` (I guess no) or can you put anything within some limits? Ideally, we should have somethin that includes `Odoo` so, one checkings the logs, knows who made those requests.
https://developers.whatismybrowser.com/learn/browser-detection/user-agents/user-agent-best-practices

Reply:
Thanks for the suggestion. I've finally tested it with a better user-agent (`Odoo (http://www.odoo.com/contactus)` and it works.